### ISSUE TITLE
Issues Using With Localstack

### ISSUE THREAD

**Issue by kevinswarner** [dynamoose/dynamoose#1534]
### Summary:

I have a simple index.js file and am attempting to work with LocalStack DynamoDB. I am probably overlooking something simple, but when I attempt to run this I get the error "CredentialsProviderError: Could not load credentials from any providers". If I get this error using the AWS DynamoDB Client directly, I can provide fake credentials (like below) and the localstack endpoint. I appreciate any help. I posted in the discussions, but have not had a response.

### Code sample:

```javascript
// index.js

const dynamoose = require("dynamoose");

const ddb = new dynamoose.aws.ddb.DynamoDB({
  accessKeyId: "FOO",
  secretAccessKey: "BAR",
  region: "us-east-1",
  endpoint: "http://localhost:4566",
});

dynamoose.aws.ddb.set(ddb);

const process = async () => {
  const schema = new dynamoose.Schema({
    id: { type: String, hashKey: true },
    message: String,
  });

  const Comment = dynamoose.model("comments", schema, { create: true });

  const comment = await Comment.create({ id: "1", message: "Hello" });

  console.log(comment);
};

process();
```

I also tried using this approach...

```javascript
// index.js

const dynamoose = require("dynamoose");

dynamoose.aws.ddb.local("http://localhost:4566");

const process = async () => {
  const schema = new dynamoose.Schema({
    id: { type: String, hashKey: true },
    message: String,
  });

  const Comment = dynamoose.model("comments", schema, { create: true });

  const comment = await Comment.create({ id: "1", message: "Hello" });

  console.log(comment);
};

process();
```

I get the following error.

CredentialsProviderError: Could not load credentials from any providers

### Environment:

Operating System: Mac OS Monterey
Operating System Version: 12.5.1
Node.js version (`node -v`): 16.15.1
NPM version: (`npm -v`): 8.11.0
Dynamoose version: 3.1.0

### Other:
- [X] I have read through the Dynamoose documentation before posting this issue
- [X] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [X] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [X] I have tested the code provided and am confident it doesn't work as intended
- [X] I have filled out all fields above
- [X] I am running the latest version of Dynamoose

**Comment by kevinswarner** [comment]
After further testing, this may simply be an issue with existing AWS credentials already being loaded somehow in the context of the terminal. When I quit the terminal and restarted, this example worked. I am unclear on the specifics, so if I get more details and can reproduce this reliably, I will create another ticket. Thanks.

**Comment by kevinswarner** [comment]
I want to add some comments here in case someone else runs into the "CredentialsProviderError: Could not load credentials from any providers" issue when running against a local DynamoDB instance. I know how I finally solved the issue in my scenarios, but I am not sure of the underlying reason using the approaches above does not always work.

For me, I have my ~/.aws/config file set up but I have no [default] section in my config. All of my interaction with "real" AWS is done through specific profiles. So, I did not have a section in my config like this...

```
[default]
aws_access_key_id = XXXXX
aws_secret_access_key = XXXXXX
```

For some reason, the AWS client gets conflicted if you do not have this AND you do not specify a profile.

So, I added in a default section with dummy values for my access key id and secret access key. I know that this default profile is only used in my scenarios for local work against a local DynamoDB (e.g. Localstack), so dummy values are fine.

Very strange and took me a long time to figure out.
